test/node/test-buffer-badhex.js

```diff
@@ -17,9 +17,10 @@ const assert = require('assert');
   // Node Buffer behavior check
   // > Buffer.from('abc def01','hex')
   // <Buffer ab>
+  assert.strictEqual(buf.write('00000000', 0, 'hex'), 4);
   assert.strictEqual(buf.write('abc def01', 0, 'hex'), 1);
-  assert.deepStrictEqual(buf, new Buffer([0xab]));
-  assert.strictEqual(buf.toString('hex'), 'ab');
+  assert.deepStrictEqual(buf, new Buffer([0xab, 0, 0, 0]));
+  assert.strictEqual(buf.toString('hex'), 'ab000000');
```

Comment:
Just replace these 4 lines with the following.

```js
  assert.deepStrictEqual(Buffer.from('abc def01', 'hex'), Buffer.from([0xab]));
```

Reply:
Probably best to keep both? To make sure that the `write` function doesn't write more than 1 byte?

Reply:
I've pushed a commit that adds your suggested line as well